network/80-container-host0-tun.network

```diff
@@ -0,0 +1,24 @@
+# SPDX-License-Identifier: MIT-0
+#
+# This config file is installed as part of systemd.
+# It may be freely copied and edited (following the MIT No Attribution license).
+#
+# To make local modifications, use "networkctl edit". See networkctl(1) for details.
+# This file should not be edited in place, because it'll be overwritten on upgrades.
+
+# This network file is inside unprivileged containers that rely on network being
+# provided as a tun link via slirp4netns(1) or pasta(1).
+
+[Match]
+Kind=tun
+Name=host0
+Virtualization=container
+
+[Network]
+DHCP=yes
+LinkLocalAddressing=yes
+LLDP=yes
+EmitLLDP=customer-bridge
```

Comment:
i figure we shouldn't do lldp for such weird userspace networking, it's not going to use that info anyway.

Reply:
I looked into these options before posting the MR. The tun device can be driven in tun mode or tap mode. In tun mode, it starts on the IP layer and LLDP does not make any sense, but it is more commonly driven in tap mode (which is what both slirp4netns and pasta do). Neither of them use LLDP as far as I understand. Another option for unprivileged (or semi-privileged) network setup is using VDE2. While VDE2 does not support LLDP by itself, it felt plausible to use it that way. In any case, I concur that currently this is likely not a use case and thus remove it.